src/lib.rs

```diff
@@ -49,6 +49,8 @@
 //! `log` crate. If not, the logs will discarded. This cargo feature is part of the default set of
 //! features of Smithay.
 
+#[doc(hidden)]
+pub extern crate io_lifetimes;
```

Comment:
Why do we need an extern crate? This isn't 2015

Reply:
I know, I just aligned it with `nix` just below, but we can also change both?

Reply:
I think `nix` and `io_lifetimes` should just be `pub use`-ed in the `reexports` module.

Reply:
I moved `io_lifetimes` to the `reexports` and removed the `pub extern crate nix`.
The `nix` crate already was in the `reexports`, but behind some feature gates. I made it available unconditonally as `nix` is not an optional crate anyway.